src/xkernel.cpp

```diff
@@ -61,11 +61,22 @@ namespace xeus
                      interpreter_ptr interpreter,
                      history_manager_ptr history_manager,
                      server_builder builder)
-        : m_config(config)
-        , m_user_name(user_name)
-        , p_interpreter(std::move(interpreter))
-        , p_history_manager(std::move(history_manager))
-        , m_builder(builder)
+        : m_config(config),
+          m_user_name(user_name),
+          p_interpreter(std::move(interpreter)),
+          p_history_manager(std::move(history_manager)),
+          m_builder(builder)
+    {
+    }
+
+    xkernel::xkernel(const std::string& user_name,
```

Comment:
We need to create the temporary json file inside of the `start` method of `xkernel`. If we don't save the configuration in a file, we have no way to retrieve the chosen port numbers from the main function.

Reply:
Or, we can split the `start` method into two methods `init_server`, and `start`. This way we can retrieve the config between the two calls, and it will be up to the kernel developer to chose whether to save the configuration in a file or not.